src/main.cc

```diff
@@ -310,6 +316,13 @@ static int daemon_main(LiteClient& client, const bpo::variables_map& variables_m
         continue;
       }
 
+      if (no_space) {
+        LOG_WARNING
+            << "Insufficient storage space was detected, skipping any updates until aktualizr-lite is restarted";
+        std::this_thread::sleep_for(std::chrono::seconds(interval));
+        continue;
```

Comment:
I'm not sure we should just stop retrying. Nobody is going to have logic to restart the daemon. Is there some way we could try and detect if free disk space had increased, so that we could start trying to install again?

Reply:
>Nobody is going to have logic to restart the daemon.

Well, I am not sure about it. Aeler kind of does it, their App does control the aklite daemon. In general, I think customers should use aklite via the API/lib or command line tools. In this case it's easier to have control over upadte, e.g. let a device user to decide when is the best time to perform an update or notify a user about insufficient space so he/she has a chance to clean a storage.


> Is there some way we could try and detect if free disk space had increased, so that we could start trying to install again?

I've thought about. I think there are many ways how to do that, but I am not that it belongs to aklite. This is up to an application that uses aklite to decide how to react on such situation, hot to clean storage. if it is so then that application can start aklite again.

In any case, the key goal of this change is to overcome the issue that Milosh discovered.

We may allow an update again when a new Target appears.

Reply:
@mike-scott @Tim-Anderson What do you think aklite should do in this case (not enough space to download update),

1. Don't keep trying to download an update until aklite is restarted or a device is rebooted.

2. Check whether available storage has increased since "not enough space" was detected and try to do an update again then.